internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -356,6 +420,175 @@ func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool)
+// value is restored.
+// Note: Passing the directories as parameters to facilitate unit tests (following the
+// pattern used by other functions in this file).
+func setCPUPMQOSResumeLatency(c *oci.Container, latency, cpuDir, cpuSaveDir string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("find container %s CPUs", c.ID())
+	}
+
+	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
+	if err != nil {
+		return err
+	}
+
+	for _, cpu := range cpus.ToSlice() {
+		if latency != "" {
+			// retrieve the current latency
+			latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
+			latencyOrig, err := os.ReadFile(latencyFile)
+			if err != nil {
+				return err
+			}
+
+			// save the current latency so we can restore it later
+			cpuPowerSaveDir := fmt.Sprintf("%s/cpu%d/power", cpuSaveDir, cpu)
+			err = os.MkdirAll(cpuPowerSaveDir, 0o750)
+			if err != nil {
+				return err
+			}
+			latencyFileOrig := path.Join(cpuPowerSaveDir, "pm_qos_resume_latency_us")
+			err = os.WriteFile(latencyFileOrig, latencyOrig, 0o644)
+			if err != nil {
+				return err
+			}
+
+			// update the pm_qos_resume_latency_us
+			err = os.WriteFile(latencyFile, []byte(latency), 0o644)
+			if err != nil {
+				return err
+			}
+		} else {
+			// retrieve the original latency
+			latencyFileOrig := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuSaveDir, cpu)
+			latencyOrig, err := os.ReadFile(latencyFileOrig)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					// the latency may have already been restored by a previous invocation of the hook
+					return nil
+				}
+				return err
+			}
+
+			// restore the original latency
+			latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
+			err = os.WriteFile(latencyFile, latencyOrig, 0o644)
+			if err != nil {
+				return err
+			}
+
+			// remove the saved latency
+			err = os.Remove(latencyFileOrig)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// setCPUFreqGovernor sets the scaling_governor for a cpu and stores the original
+// value so it can be restored later. If the governor is an empty string, the original
+// scaling_governor value is restored.
+// Note: Passing the directories as parameters to facilitate unit tests (following the
+// pattern used by other functions in this file).
+func setCPUFreqGovernor(c *oci.Container, governor, cpuDir, cpuSaveDir string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("find container %s CPUs", c.ID())
+	}
+
+	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
+	if err != nil {
+		return err
+	}
+
+	for _, cpu := range cpus.ToSlice() {
+		if governor != "" {
```

Comment:
It looks like `governorFile` and `governorFileOrig` are used in both branches (`if` and `else`), so it makes sense to move them out of the block. IOW:

```go
	for _, cpu := range cpus.ToSlice() {
		governorFile := fmt.Sprintf("%s/cpu%d/cpufreq/scaling_governor", cpuDir, cpu)
		cpuFreqSaveDir := fmt.Sprintf("%s/cpu%d/cpufreq", cpuSaveDir, cpu)
		governorFileOrig := path.Join(cpuFreqSaveDir, "scaling_governor")
		if governor != "" {
			....
```

This way we are sure we're using the same files on set and restore.

Reply:
Done - good suggestion.